subsys/net/ip/icmpv4.c

```diff
@@ -208,6 +210,7 @@ int net_icmpv4_send_echo_request(struct net_if *iface,
 
 	echo_req->identifier = htons(identifier);
 	echo_req->sequence   = htons(sequence);
+	memcpy(echo_req->data, data, data_size);
```

Comment:
Even though, at this stage of this function implementation, this logic is not wrong, it won't fly with time if one day the icmpv4_access is not a contiguous one anymore.

And in any case, never memcpy (or memmove  etc..) by yourself. Alway use the net_pkt r/w api, because that API can tell you if it could r/w or not (it verifies that buffer does have the necessary space)

So after net_pkt_set_data() below, just do a net_pkt_write(pkt, data, data_size)

Reply:
Ok, changed it to net_pkt_write_new().